app/core/Tracker2/FullStoryTracker.js

```diff
@@ -0,0 +1,175 @@
+  }
+
+  async _initializeTracker () {
+    this.watchForDisableAllTrackingChanges(this.store)
+
+    // TODO handle disable all tracking
+    window['_fs_ready'] = () => {
+      try {
+        this.log('ready')
+        if ((new URLSearchParams(window.location.search)).has(FULLSTORY_ENABLE_QUERY_PARAM)) {
+          this.enabled = true
+          this.log('query param force enable')
+        } else if (!this.enableDecisionMade) {
+          this.log('deciding on enabled')
+          this.enabled = this.decideEnabled()
+          if (this.enabled) {
+            this.globalTracker.trackEvent('FullStory Tracking Enabled')
+          }
+        }
+
+        if (this.enabled) {
+          this.log('enabling from init')
+          this.enable()
+        } else {
+          this.log('disabling from init')
+          this.disable()
+        }
+
+        this.onInitializeSuccess()
+      } catch (e) {
+        this.onInitializeFail(e)
+        throw e
+      }
+    }
+
+    loadFullStory()
+  }
+
+  enable () {
+    this.enabled = true
+    window.sessionStorage.setItem(FULLSTORY_SESSION_TRACKING_ENALBED_KEY, 'true')
+    FS.restart()
+    this.log('enabled')
+  }
+
+  disable () {
+    this.enabled = false
+    window.sessionStorage.setItem(FULLSTORY_SESSION_TRACKING_ENALBED_KEY, 'false')
+    FS.shutdown()
+    this.log('disabled')
+  }
+
+  decideEnabled () {
+    const { me } = this.store.state
+
+    if (this.disableAllTracking) {
+      this.log('decide enabled', 'disable all tracking')
+      return false
+    } else if (me.anonymous && Math.random() < 0.02) {
+      this.log('decide enabled', 'anon user')
+      return true
+    } else if (this.store.getters['me/isTeacher'] && !this.store.getters['me/isParent'] && Math.random() < 0.02) {
+      this.log('decide enabled', 'non parent teacher')
+      return true
+    }
+
+    this.log('decide enabled', 'not enabled')
+    return false
+  }
+
+  async identify (traits = {}) {
+    await this.initializationComplete
+
+    // When identify() is called it's possible that the user has changed, so check if the user
+    // has changed and if so run the decision logic again
+    const { me } = this.store.state
+    const lastUserId = window.sessionStorage.getItem(FULLSTORY_LAST_USER_ID_KEY)
+    if (lastUserId !== me._id) {
+      this.log('identify', 'new user')
+      this.enabled = this.decideEnabled()
+      if (this.enabled) {
+        this.log('identify', 'enabling')
+        this.enable()
+      } else {
+        this.log('identify', 'disabling')
+        this.disable()
+      }
+    }
+
+    window.sessionStorage.setItem(FULLSTORY_LAST_USER_ID_KEY, me._id)
+    if (!this.enabled) {
+      return
+    }
+
+    FS.identify(me._id)
+    FS.setUserVars(extractDefaultUserTraits(me))
+  }
+
+  async trackPageView (includeIntegrations = []) {
+    await this.initializationComplete
+
+    if (!this.enabled) {
+      return
+    }
+
+    const url = `/${Backbone.history.getFragment()}`
```

Comment:
Looks good! :) Would it ever be useful to use a browser api: `location.pathname`

Reply:
Hm yeah maybe we should switch to this, also going to capture as a task to investigate.  Pulled this from original tracker.

Reply:
https://app.asana.com/0/1155700512434711/1177758721954643/f